### ISSUE TITLE
installinstallmacos.py Checksum check failing

### ISSUE THREAD

**Issue by Rorzzer** [grahampugh/erase-install#196]
Freshly downloaded version of installinstallmacos.py fails the shasum check with error,
/Library/Management/erase-install/installinstallmacos.py: FAILED
shasum: WARNING: 1 computed checksum did NOT match
   [get_installinstallmacos] ERROR: downloaded installinstallmacos.py does not match checksum. Possible corrupted file. Deleting file.

**To Reproduce**
Run the following command
curl -s https://raw.githubusercontent.com/grahampugh/erase-install/main/erase-install.sh | sudo bash /dev/stdin --reinstall --os=12 --update --depnotify --cleanup-after-use --check-power --force-curl --power-wait-limit 180

or
Upload package to JAMF and deploy then run command via policy,
/Library/Management/erase-install/erase-install.sh --reinstall --os=12 --update --depnotify --cleanup-after-use --check-power --force-curl --power-wait-limit 180

Have tried via terminal and JAMF, the issue is the same. Installed the package via JAMF and also tried downloading fresh from github and installing manually. Same result. 

Manually checking checksum result,
shasum -a 256 /Library/Management/erase-install/installinstallmacos.py
08ceb0187bd648e040c8ba23f79192f7d91b1250dbff47107c29cb2bca1ce433  /Library/Management/erase-install/installinstallmacos.py

Checksum hardcoded into eraseinstall.sh
08ceb0187bd648e040c8ba23f79192f7d91b1250dbff47107c29cb2bca1ce433
 
**Expected behavior**

Checksum to be verified as matching, install to continue as usual. .py file not to be deleted. 

**Code/log output**
Please supply the full command used, and if applicable, add full output from Terminal or from `/var/log/erase-install.log`. Either upload the log, or paste the output in a code block (triple backticks at the start and end of the code block please!).

JAMF Log
```
Running command /Library/Management/erase-install/erase-install.sh --reinstall --os=12 --update --depnotify --cleanup-after-use --check-power --force-curl --power-wait-limit 180...
--
Result of command:[erase-install] v25.0 script execution started: Thu Feb  3 15:06:08 AEDT 2022    [erase-install] Caffeinating this script (pid=4120)    [check_free_space] OK - 159 GB free/purgeable disk space detected    [check_power_status] OK - AC power detected    [erase-install] Looking for existing installer app or pkg    [find_existing_installer] No valid installer found.    [erase-install] Opening DEPNotify download message (language=en)    [get_installinstallmacos] Downloading installinstallmacos.py... /Library/Management/erase-install/installinstallmacos.py: FAILED shasum: WARNING: 1 computed checksum did NOT match    [get_installinstallmacos] ERROR: downloaded installinstallmacos.py does not match checksum. Possible corrupted file. Deleting file. Could not download installinstallmacos.py so cannot continue.    [erase-install] 'caffeinate' ended /Library/Management/erase-install/erase-install.sh: line 945:  4131 Terminated: 15          /usr/bin/caffeinate -dimsu -w $pid /Library/Management/erase-install/erase-install.sh: line 724:  4159 Killed: 9               dep_notify_progress installinstallmacos > /dev/null 2>&1
```

erase-install.log
```
   [erase-install] Caffeinating this script (pid=6142)
   [check_free_space] OK - 165 GB free/purgeable disk space detected
   [check_power_status] OK - AC power detected
   [erase-install] Looking for existing installer app or pkg
   [find_existing_installer] No valid installer found.
   [erase-install] Opening DEPNotify download message (language=en)
   [get_installinstallmacos] Downloading installinstallmacos.py...
/Library/Management/erase-install/installinstallmacos.py: FAILED
shasum: WARNING: 1 computed checksum did NOT match
   [get_installinstallmacos] ERROR: downloaded installinstallmacos.py does not match checksum. Possible corrupted file. Deleting file.
Could not download installinstallmacos.py so cannot continue.
   [erase-install] 'caffeinate' ended
/dev/stdin: line 949:  6155 Terminated: 15          /usr/bin/caffeinate -dimsu -w $pid
/dev/stdin: line 728:  6182 Killed: 9               dep_notify_progress installinstallmacos > /dev/null 2>&1
```

**Screenshots**

**Environment (please complete the following information):**
 - OS version [e.g. 12.0.1] - 12.0.1
 - erase-install version [e.g. 24.0] - 25.0

**Additional context**

**Comment by Rorzzer** [comment]
Tried this on another device, brand new out of the box, freshly enrolled. macOS 12.1. Same error.

```
Running command /Library/Management/erase-install/erase-install.sh --reinstall --os=12 --update --depnotify --cleanup-after-use --check-power --force-curl --power-wait-limit 180...
--
Result of command:[erase-install] v25.0 script execution started: Thu Feb  3 17:06:08 AEDT 2022    [erase-install] Caffeinating this script (pid=10927)    [check_free_space] OK - 960 GB free/purgeable disk space detected    [check_power_status] WARNING - No AC power detected    [wait_for_power] Waiting for AC power...    [wait_for_power] OK - AC power detected    [erase-install] 'jamfHelper' ended    [erase-install] Looking for existing installer app or pkg    [find_existing_installer] No valid installer found. /Library/Management/erase-install/erase-install.sh: line 1636: 10962 Terminated: 15          "$jamfHelper" -windowType "utility" -title "${!dialog_power_title}" -description "${!dialog_power_desc} ${power_wait_timer_friendly}" -alignDescription "left" -icon "$dialog_confirmation_icon"    [erase-install] Opening DEPNotify download message (language=en)    [get_installinstallmacos] Downloading installinstallmacos.py... /Library/Management/erase-install/installinstallmacos.py: FAILED shasum: WARNING: 1 computed checksum did NOT match    [get_installinstallmacos] ERROR: downloaded installinstallmacos.py does not match checksum. Possible corrupted file. Deleting file. Could not download installinstallmacos.py so cannot continue.    [erase-install] 'caffeinate' ended /Library/Management/erase-install/erase-install.sh: line 945: 10938 Terminated: 15          /usr/bin/caffeinate -dimsu -w $pid /Library/Management/erase-install/erase-install.sh: line 724: 11043 Killed: 9               dep_notify_progress installinstallmacos > /dev/null 2>&1
```

**Comment by grahampugh** [comment]
@Rorzzer apologies, I forgot about the checksum check. I've now reverted the change to `installinstallmacos.py` in its `main` branch.

**Comment by AirBookMac** [comment]
Hey @grahampugh, we have the same issue since today. We are using erase-install-depnotify-25.0.pkg 

Running command is 
`/Library/Management/erase-install/erase-install.sh --os=12 --erase --current-user --confirm --depnotify --check-power --force-curl --update`
and Result command is:
`Result of command:
[erase-install] v25.0 script execution started: Thu Feb  3 11:00:50 CET 2022
   [erase-install] Caffeinating this script (pid=3678)
   [check_free_space] OK - 456 GB free/purgeable disk space detected
   [check_power_status] OK - AC power detected
   [erase-install] Looking for existing installer app or pkg
   [find_existing_installer] No valid installer found.
yes username is a member of everyone
   [get_user_details] username is a Volume Owner
   [check_password] Success: the password entered is the correct login password for username.
no username is NOT a member of admin
   [get_user_details] username account has been promoted to admin so that eraseinstall can proceed
   [erase-install] Opening DEPNotify download message (language=de)
   [get_installinstallmacos] Downloading installinstallmacos.py...
/Library/Management/erase-install/installinstallmacos.py: FAILED
shasum: WARNING: 1 computed checksum did NOT match
   [get_installinstallmacos] ERROR: downloaded installinstallmacos.py does not match checksum. Possible corrupted file. Deleting file.
Could not download installinstallmacos.py so cannot continue.
   [erase-install] 'caffeinate' ended
/Library/Management/erase-install/erase-install.sh: line 945:  3689 Terminated: 15          /usr/bin/caffeinate -dimsu -w $pid
/Library/Management/erase-install/erase-install.sh: line 724:  3735 Killed: 9               dep_notify_progress installinstallmacos > /dev/null 2>&1
   [erase-install] 'DEPNotify' ended
     [erase-install] User username was demoted back to standard user`

**Comment by grahampugh** [comment]
Hi folks, please try again with 25.0. I uploaded "new" versions of it at around 17:00 UTC which should hopefully solve your issue. Sorry for the inconvenience.

**Comment by AirBookMac** [comment]
hi. it works like a charme with the new version. 👍  thx a lot!

**Comment by Rorzzer** [comment]
> Hi folks, please try again with 25.0. I uploaded "new" versions of it at around 17:00 UTC which should hopefully solve your issue. Sorry for the inconvenience.

Hi @grahampugh 

I'm still seeing the checksum error, I tried with the "new" 25.0 release and I've also tried with the 26.0 Pre-Release. 

```


Installing erase-install-depnotify-26.0.pkg...
--
Successfully installed erase-install-depnotify-26.0.pkg.
[STEP 3 of 5]
[STEP 4 of 5]
Running command /Library/Management/erase-install/erase-install.sh --reinstall --os=12 --update --depnotify --cleanup-after-use --check-power --force-curl --power-wait-limit 180...
Result of command:[erase-install] v26.0 script execution started: Tue Feb  8 14:17:04 AEDT 2022    [erase-install] Caffeinating this script (pid=30561)    [check_free_space] OK - 893 GB free/purgeable disk space detected    [check_power_status] OK - AC power detected    [erase-install] Looking for existing installer app or pkg    [find_existing_installer] No valid installer found.    [erase-install] Opening DEPNotify download message (language=en) Adding DEPNotify button OK    [get_installinstallmacos] Downloading installinstallmacos.py... /Library/Management/erase-install/installinstallmacos.py: FAILED shasum: WARNING: 1 computed checksum did NOT match    [get_installinstallmacos] ERROR: downloaded installinstallmacos.py does not match checksum. Possible corrupted file. Deleting file. Could not download installinstallmacos.py so cannot continue.    [erase-install] 'caffeinate' ended /Library/Management/erase-install/erase-install.sh: line 767: 30604 Killed: 9               dep_notify_progress installinstallmacos > /dev/null 2>&1

```

**Comment by Rorzzer** [comment]
Sorry, the "new" 25.0 package actually works fine (something was caching on my end), it is actually just the 26.0 pre-release that still fails the checksum. I understand that this is a pre-release for a reason. Just a heads up!

**Comment by grahampugh** [comment]
Gah, choosing the right checksum is hard! I think I fixed it now in v26.0.

**Comment by AirBookMac** [comment]
> Sorry, the "new" 25.0 package actually works fine (something was caching on my end), it is actually just the 26.0 pre-release that still fails the checksum. I understand that this is a pre-release for a reason. Just a heads up!

Hi, we had to delete /Library/Management/erase-install and then the updated 25.0 worked again.
